toolkit/tools/imagegen/diskutils/diskutils.go

```diff
@@ -803,12 +804,12 @@ func FormatSinglePartition(partDevPath string, partition configuration.Partition
 			fsType = "vfat"
 		}
 
-		mkfsArgs := []string{"-t", fsType}
+		mkfsArgs := []string{"--timeout", timeoutInSeconds, diskDevPath, "mkfs", "-t", fsType}
```

Comment:
Doesn't need to be here, but I wonder if its worth having a dedicated flock function that wraps shell.Execute (or maybe an extra bit of logic on execute?).

Reply:
I have thought about potentially creating a helper type that calls the file lock API directly, instead of using `flock`. Though such a change would be rather involved. So, I am not sure how worth it it would be, seeing that using `flock` doesn't seem to cause any problems other than making the code slightly harder to read.